src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/InMemoryAssemblyLoader.cs

```diff
@@ -20,6 +24,9 @@ public static class InMemoryAssemblyLoader
         public static unsafe void LoadInMemoryAssembly(IntPtr moduleHandle, IntPtr assemblyPath)
         {
 #if TARGET_WINDOWS
+            if (!IsSupported)
+                throw new NotSupportedException("This API is not enabled in trimmed scenarios. see https://aka.ms/dotnet-illink/nativehost for more details");
+
```

Comment:
If I read this correctly the warning in this case will come from the constructor of `IsolatedComponentLoadContext` - which doesn't have the correct message and URL on it. We should either create a new method which will provide the right warning, or fix the constructor.